main.c

```diff
@@ -1175,6 +1187,19 @@ int main(int argc, char **argv) {
       // Otherwise, we're still alive. Re-check next time.
     }
 
+    if (signal_wakeup) {
+      // A signal was received to request a wakeup. Clear the flag
+      // and proceed to auth.
+      signal_wakeup = 0;
```

Comment:
FYI this is technically a race condition - between check and setting to 0 the signal may come in again.

Please add a comment that we accept this race, as waking up the screen is idempotent.